packages/commonwealth/server/routes/updateChain.ts

```diff
@@ -118,6 +119,52 @@ const updateChain = async (
     return next(new AppError(Errors.InvalidTerms));
   }
 
+  const snapshotSpaces = await models.CommunitySnapshotSpaces.findAll({
+    where: { chain_id: chain.id },
+    include: {
+      model: models.SnapshotSpace,
+      as: 'snapshot_space',
+    },
+  });
+
+  // Check if any snapshot spaces are being removed
+  const removedSpaces = snapshotSpaces.filter((space) => {
+    return !snapshot.includes((space as any).snapshot_space.snapshot_space);
+  });
+  const existingSpaces = snapshotSpaces.filter((space) => {
+    return snapshot.includes((space as any).snapshot_space.snapshot_space);
+  });
+  const existingSpaceNames = existingSpaces.map((space) => {
+    return (space as any).snapshot_space.snapshot_space;
+  });
+
+  for (const spaceName of snapshot) {
+    // check if its in the mapping
+    if (existingSpaceNames.includes(spaceName)) {
+      continue;
+    } else {
+      const spaceModelInstance = await models.SnapshotSpace.findOrCreate({
```

Comment:
Any validation that the space exists?

Reply:
we should, given that next step is to hook listeners up to snapshots -- could be as easy as hitting a snapshot endpoint and verifying the space's metadata. we can also think about what data we want to store in the db / cache.

Reply:
@rvmelkonian Going to wait on this until we have decided how this will work snapshot-listener side, probably some logic to reuse